internals/src/slice.rs

```diff
@@ -0,0 +1,116 @@
+#![allow(unstable_name_collisions)]
+//! Contains extensions related to slices.
+
+/// Extension trait for slice.
+pub trait SliceExt {
+    /// The item type the slice is storing.
+    type Item;
+
+    /// Splits up the slice into a slice of arrays and a remainder.
+    ///
+    /// Note that `N` must not be zero:
+    ///
+    /// ```compile_fail
+    /// let slice = [1, 2, 3];
+    /// let fail = slice.as_chunks::<0>();
+    /// ```
+    fn as_chunks<const N: usize>(&self) -> (&[[Self::Item; N]], &[Self::Item]);
+
+    /// Splits up the slice into a slice of arrays and a remainder.
+    ///
+    /// Note that `N` must not be zero:
+    ///
+    /// ```compile_fail
+    /// let mut slice = [1, 2, 3];
+    /// let fail = slice.as_chunks_mut::<0>();
+    /// ```
+    fn as_chunks_mut<const N: usize>(&mut self) -> (&mut [[Self::Item; N]], &mut [Self::Item]);
+}
+
+impl<T> SliceExt for [T] {
+    type Item = T;
+
+    fn as_chunks<const N: usize>(&self) -> (&[[Self::Item; N]], &[Self::Item]) {
+        //let _ = Hack::<N>::IS_NONZERO;
```

Comment:
Forgotten comment?

Reply:
Thanks, it was supposed to be uncommented, I was just looking what it does when it's commented. :D